metabase/metabase
# Set case-insensitive option for template tags

Epic https://github.com/metabase/metabase/issues/29881

Makes **new** string field filters in native queries take default operator options into account. This makes `string/contains`, `string/does-not-contain`, `string/starts-with`, `string/ends-with` case-insensitive. Existing parameters won't be affected until the user changes the type manually.

How to verify:
- New -> SQL -> `SELECT * FROM PRODUCTS WHERE {{f}}`
- Text -> Field Filter -> Products -> Title
- The widget type should be set to `String contains`
- Set parameter value to `bronze` and run the query. It should find `Incredible Bronze Pants`.
- Remove the parameter value and set the same value as the default one and run the query. There should be the same result.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30121)
